### PR TITLE
Python: Honor the function calling options if configured.

### LINKED ISSUE #5661: Python: Function Calling Stepwise Planner - Max Tokens Ignored after First Call
**Describe the bug**
I've been experimenting with the Function Calling Stepwise Planner in python (really awesome, thanks for implementing this). I noticed that the first chat completion call respects the max_completion_tokens setting given in the FunctionCallingStepwisePlannerOptions config. However, all the subsequent calls seem to pull from the default from OpenAIPromptExecutionSettings and for me this meant it defaulted back to 256. Function calling seems to consistently require more tokens for its response so this results in responses that terminate with a finish_reason of 'length' and the function call isn't in the right format and things break. Experimenting locally I tweaked the default OpenAIPromptExecutionSettings's max_tokens and I could see that this is what is being followed after the first chat completion. 

**To Reproduce**
1. Use the Function Calling Stepwise Planner with a max_completion_tokens setting other than 256. 
2. Execute any ask (for me I even saw this when the ask was 'hi')
3. Look at the logs, observe the first and subsequent max_token settings. 
4. Observe the max_completion_tokens setting only honored on the first call.

**Expected behavior**
FunctionCallingStepwisePlannerOptions's max_completion_tokens followed for all calls.

**Screenshots**
N/A

**Platform**
 - OS: Mac
 - IDE: Vs Code
 - Language:Python
 - Source: main branch of this repo

**Additional context**
If I have time I can try to investigate further, but right now this is the best I can do!
I was using gpt4 for these tests.

### PR DESCRIPTION
**Pull request by moonbox3**
### Motivation and Context

In the function calling stepwise planner, some settings like max_tokens (based on max_completion_tokens from the planner settings) and the max auto invoke attempts were not used.

### Description

This PR fixes the issue where the prompt execution settings for chat completion wasn't using the function calling stepwise planner options, if configured. Closes #5661

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/10f38e4b64a3544970a37ffa3dc43ab3b357fc36/README.md">Py3.8 Test Coverage</a><details><summary>Python 3.8 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel/planners/function_calling_stepwise_planner</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/10f38e4b64a3544970a37ffa3dc43ab3b357fc36/python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py">function_calling_stepwise_planner.py</a></td><td>94</td><td>36</td><td>62%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/10f38e4b64a3544970a37ffa3dc43ab3b357fc36/python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py#L113-L171">113&ndash;171</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/10f38e4b64a3544970a37ffa3dc43ab3b357fc36/python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py#L202-L207">202&ndash;207</a></td></tr><tr><td><b>TOTAL</b></td><td><b>5398</b></td><td><b>1007</b></td><td><b>81%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python 3.8 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1200 | 11 :zzz: | 0 :x: | 0 :fire: | 24.011s :stopwatch: |

**Review by eavanvalkenburg — APPROVED** [review]

File: python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner_options.py [hunk 1]
```diff
@@ -13,15 +15,15 @@
...
-    max_tokens: Optional[int] = None
-    max_tokens_ratio: Optional[float] = 0.1
-    max_completion_tokens: Optional[int] = None
-    max_prompt_tokens: Optional[int] = None
-    get_initial_plan: Optional[Callable[[], str]] = None
-    get_step_prompt: Optional[Callable[[], str]] = None
-    max_iterations: Optional[int] = 15
-    min_iteration_time_ms: Optional[int] = 100
-    execution_settings: Optional[OpenAIPromptExecutionSettings] = None
+    max_tokens: int | None = None
```
Comment (juliomenendez) [inline python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner_options.py:18]: Is `Optional` no longer the way to do this?

**Review by juliomenendez — APPROVED** [review]

File: python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner_options.py (hunk 1 above)
Comment (moonbox3) [reply python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner_options.py:18]: Now that we're starting to use `from __future__ import annotations` in the codebase, we get the nicety of being able to use the newer type hints. In Python 3.8/3.9, type hints like `int | None` are treated as string literals, and their evaluations are deferred. Python 3.8/3.9 will see the `|` operator as a union type, similar to Optional[int].

Comment (MediVik333) [reply python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner_options.py:18]: Ok

**Review by MediVik333 — COMMENTED** [review]
Save

**Review by juliomenendez — APPROVED** [review]

**Comment by MediVik333** [comment]
Thanks

**Pull request merged**